### PR TITLE
Page source

### PR DESCRIPTION
**Pull request by shs96c**
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/6810)

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None


<h2>Unstable Results</h2>
  
  
  
  <h3>Browser: "Firefox Nightly"<small></small></h3>
  <p>View in: <a href="http://pulls.web-platform-tests.org/job/17234.3">Dashboard</a> |
      <a href="https://travis-ci.org/w3c/web-platform-tests/jobs/263213042">TravisCI</a></p>
  <table>
    <tr>
      <th>Test</th>
      <th>Subtest</th>
      <th>Results</th>
      <th>Messages</th>
    </tr>
  
  
  
  
  
  
  <tr>
    <td><code>/webdriver/tests/navigation/current_url.py</code></td>
    <td>&nbsp;</td>
    <td>OK: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_get_current_url_after_modified_location</code></td>
    <td>PASS: 8<br />FAIL: 2<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_get_current_url_alert_prompt</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_get_current_url_file_protocol</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_get_current_url_matches_location</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_get_current_url_nested_browsing_context</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_get_current_url_nested_browsing_contexts</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_get_current_url_no_browsing_context</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_get_current_url_payload</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_get_current_url_special_pages</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>current_url.py::test_set_malformed_url</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td><code>/webdriver/tests/navigation/get_title.py</code></td>
    <td>&nbsp;</td>
    <td>OK: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_after_modification</code></td>
    <td>PASS: 9<br />FAIL: 1<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_from_closed_context</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_from_frame</code></td>
    <td>FAIL: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_from_top_context</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_handle_prompt_accept</code></td>
    <td>FAIL: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_handle_prompt_dismiss</code></td>
    <td>FAIL: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_handle_prompt_missing_value</code></td>
    <td>FAIL: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_strip_and_collapse</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_with_duplicate_element</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  <tr>
    <td>&nbsp;</td>
    <td><code>get_title.py::test_title_without_element</code></td>
    <td>PASS: 10<br /></td>
    <td></td>
  </tr>
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  
  </table>

**Review by andreastt — APPROVED** [review]

**Comment by andreastt** [comment]
Not exactly sure why the CI is failing here.

**Comment by w3c-bots** [comment]
# Build ERRORED

Started: 2017-08-18 19:06:42
Finished: 2017-08-18 19:15:42


<h2>Failing Jobs</h2>
<ul>

<li>firefox:nightly</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/17554)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/266015253)

File: webdriver/tests/document_handling/page_source.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+import json
+
+from tests.support.asserts import assert_error, assert_success, assert_dialog_handled
+from tests.support.fixtures import create_dialog
+from tests.support.inline import inline
+
+
+def test_closed_browsing_context(new_session):
+    _, session = new_session({})
+    session.end()
```
Comment (andreastt) [inline webdriver/tests/document_handling/page_source.py]: You want `create_window` and then get the page source from the closed browsing context. That will return `no such window`. This is testing that you can’t get the page source of an ended session, which would return `session not created`.

**Review by andreastt — APPROVED** [review]

**Comment by gsnedders** [comment]
@andreastt is this okay to land then?

**Review by andreastt — APPROVED** [review]

**Comment by andreastt** [comment]
@gsnedders: Yes, thanks for reminding me!

**Pull request merged**